### PR TITLE
Changes to Table/Table Row for improved performance

### PR DESCRIPTION
**Pull request by dangkevin**
### Jira Ticket:
https://kajabi.atlassian.net/browse/MAN-1058

This PR adds improved performance to the Table/Table Row Sage Components in reducing number of renders, some clean up and adding some unit tests.

### CODE REVIEW

File: packages/sage-react/lib/Table/Table.jsx [hunk 1]
```diff
@@ -64,10 +64,6 @@ export const Table = ({
     }
   );
 
-  //
```
Comment (philschanely) [inline packages/sage-react/lib/Table/Table.jsx:67]: Any reason this is removed?

File: packages/sage-react/lib/Table/Table.jsx [hunk 2]
```diff
@@ -152,40 +148,31 @@ export const Table = ({
     buildHeaders();
   }, [schema, headers]);
 
-  // Ensure selected rows change when adjsuted from the outside
+  // Ensure selected rows change when adjusted from the outside
```
Comment (philschanely) [inline packages/sage-react/lib/Table/Table.jsx:151]: Thanks! ❤️

File: packages/sage-react/lib/Table/Table.jsx [hunk 3]
```diff
@@ -152,40 +148,31 @@ export const Table = ({
...
   useEffect(() => {
     setSelfSelectedRows(selectedRows);
   }, [selectedRows]);
 
+  const removeFromList = (data) => selfSelectedRows.filter((each) => each !== data);
```
Comment (philschanely) [inline packages/sage-react/lib/Table/Table.jsx:156]: Nice and minified. 🙌

**Review by philschanely — APPROVED** [review]
Looks good; one small deletion would be nice to restore if possible.

File: packages/sage-react/lib/Table/Table.jsx (hunk 1 above)
Comment (dangkevin) [reply packages/sage-react/lib/Table/Table.jsx:67]: sorry let me re-add that, accidental delete

**Review by kimjaej — APPROVED** [review]
LGTM 👍  Good job

**Comment by dangkevin** [comment]
Can we get just one more approval from @Kajabi/ux-devs 😀

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM, nice work!

**Pull request merged**